src/fields/field_fe.cc

```diff
@@ -687,15 +682,17 @@ void FieldFE<spacedim, Value>::calculate_native_values(ElementDataCache<double>:
 	data_vec_.zero_entries();
 	VectorMPI::VectorDataPtr data_vector = data_vec_.data_ptr();
 	std::vector<LongIdx> global_dof_indices(dh_->max_elem_dofs());
+	std::vector<int> const & el_ids = ReaderCache::get_reader(reader_file_)->get_element_vector(this->boundary_domain_);
 
 	// iterate through cells, assembly MPIVector
 	for (auto cell : dh_->own_range()) {
 		dof_size = cell.get_dof_indices(global_dof_indices);
 		LocDofVec loc_dofs = cell.get_loc_dof_indices();
-		data_vec_i = cell.elm_idx() * dof_size;
+		auto pos = std::find(el_ids.begin(), el_ids.end(), source_target_mesh_elm_map_[cell.elm_idx()]) - el_ids.begin();
```

Comment:
Change meaning of `source_target_mesh_elm_map_` it should map element index in the calculation mesh to the row index in the element cache